From 253024037da66c1d412b3d3064df663eda753c19 Mon Sep 17 00:00:00 2001
From: Scott Theisen <scott.the.elm@gmail.com>
Date: Fri, 19 Dec 2025 21:07:18 -0500
Subject: [PATCH] Enable VAAPI deinterlacers for AMD GPUs

Without this change, using an AMD Radeon RX 9060 XT, FFmpeg
gives the following error and will not create any VAAPI surfaces
for the deinterlacer:
Failed to create surface: 10 (attribute not supported).

Limit the workarounds to the specific driver they are for (i965).
---
 mythtv/libs/libmythtv/decoders/mythvaapicontext.cpp | 9 +--------
 1 file changed, 1 insertion(+), 8 deletions(-)

diff --git a/mythtv/libs/libmythtv/decoders/mythvaapicontext.cpp b/mythtv/libs/libmythtv/decoders/mythvaapicontext.cpp
index 978da49f6c8..e374fb9643c 100644
--- a/mythtv/libs/libmythtv/decoders/mythvaapicontext.cpp
+++ b/mythtv/libs/libmythtv/decoders/mythvaapicontext.cpp
@@ -295,14 +295,7 @@ int MythVAAPIContext::InitialiseContext(AVCodecContext* Context)
     // This may need extending for AMD etc
 
     auto vendor = interop->GetVendor();
-    // Intel NUC
-    if (vendor.contains("iHD", Qt::CaseInsensitive) && vendor.contains("Intel", Qt::CaseInsensitive))
-    {
-        vaapi_frames_ctx->attributes = nullptr;
-        vaapi_frames_ctx->nb_attributes = 0;
-    }
-    // i965 series
-    else
+    if (vendor.contains("i965", Qt::CaseInsensitive))
     {
         int format = VA_FOURCC_NV12;
         if (vendor.contains("ironlake", Qt::CaseInsensitive))
